src/Access/tests/gtest_access_rights_ops.cpp

```diff
@@ -1,8 +1,230 @@
 #include <gtest/gtest.h>
 #include <Access/AccessRights.h>
+#include <Access/AccessRights.cpp>  // NOLINT(bugprone-suspicious-include)
+#include <IO/WriteBufferFromString.h>
+
+#include <list>
 
 using namespace DB;
 
+TEST(AccessRights, Radix)
```

Comment:
I believe we need a test like this but the following list of very simple grants:

- GRANT USAGE ON *.* (nothing is granted)
- GRANT SELECT ON *.*
- GRANT SELECT ON db.*
- GRANT SELECT ON db.table
- GRANT SELECT(col1, col2) ON db.table
- GRANT SELECT ON db.*, INSERT ON db.table
- GRANT SELECT ON db.*, REVOKE SELECT ON db.table
- GRANT SELECT ON db.table WITH GRANT OPTION

Then with wildcards:
- GRANT SELECT ON db*
- GRANT SELECT ON db*, REVOKE SELECT ON dba*
- GRANT SELECT ON db*, REVOKE SELECT ON dba
- GRANT SELECT ON db, REVOKE SELECT ON dba*
- GRANT SELECT ON db*, REVOKE SELECT ON db

Without that it's very difficult to see how those trees work now.